swarm/network/kademlia.go

```diff
@@ -451,11 +451,14 @@ func (k *Kademlia) neighbourhoodDepth() (depth int) {
 	var lastPo int
 
 	f := func(v pot.Val, i int) bool {
+		if bytes.Equal(pot.ToBytes(v), pivotAddr) {
```

Comment:
Under what condition should we be getting ourselves in these iterations. Isn't that actually an anomaly?

Reply:
@nolash this is exactly the same as was removed on the left side L587:
```
			if po == 256 {	
```
...
@zelig am I correct?

Reply:
maybe going back to `po == 256` is simpler

Reply:
@justelad the question is about the iterator, not the value itself. Should the iterator _ever_ return ourselves?

Reply:
good question. i guess _not_ since the method name is `EachNeighbour` and `self` is not a neighbour. i'll make the change. good point 👍

Reply:
@nolash, come to think of it, that method name could be actually changed to `EachPeer`, since it refers to actual peers connected, not just neighbours.
@zelig, I noted to @nolash that some of the terminology in the kademlia is somewhat misleading (as I already told you about `minProxBinSize`), `EachNeighbour` is the same case - we iterate over peers not neighbours, the iterator in this case should not have a notion of what is neighbouring and what is not, and that is up to the iteratee to assert. (my 2 cents)